Sources/Flatten.swift

```diff
@@ -383,7 +383,18 @@ extension SignalProducer {
 	/// - returns: A producer that will start `self` and then on completion of
 	///            `self` - will start `next`.
 	public func concat(_ next: SignalProducer<Value, Error>) -> SignalProducer<Value, Error> {
-		return SignalProducer<SignalProducer<Value, Error>, Error>([ self.producer, next ]).flatten(.concat)
+		return SignalProducer<SignalProducer<Value, Error>, Error>([ self, next ]).flatten(.concat)
+	}
+	
+	/// `concat`s `next` onto `self`.
+	///
+	/// - parameters:
+	///   - next: A follow-up producer to concat `self` with.
+	///
+	/// - returns: A producer that will start `self` and then on completion of
+	///            `self` - will start `next`.
+	public func concat<Next: SignalProducerConvertible>(_ next: Next) -> SignalProducer<Value, Error> where Next.Value == Value, Next.Error == Error {
+		return concat(next.producer)
 	}
```

Comment:
Why do we need 2 versions of all these operators?

Reply:
To disambiguate when contextual lookup is used. Context: https://github.com/ReactiveCocoa/ReactiveSwift/pull/611#discussion_r170457763